src/Config/DataSource.cs

```diff
@@ -1,127 +1,41 @@
-    /// <param name="ConnectionString">The ADO.NET connection string that runtime
-    /// will use to connect to the backend database.</param>
-    public record DataSource(
-        [property: JsonPropertyName(DataSource.DATABASE_PROPERTY_NAME)]
-        DatabaseType DatabaseType,
-        [property: JsonPropertyName(DataSource.OPTIONS_PROPERTY_NAME)]
-        object? DbOptions = null)
+    public TOptionType? GetTypedOptions<TOptionType>() where TOptionType : IDataSourceOptions
     {
-        public const string JSON_PROPERTY_NAME = "data-source";
-        public const string DATABASE_PROPERTY_NAME = "database-type";
-        public const string CONNSTRING_PROPERTY_NAME = "connection-string";
-        public const string OPTIONS_PROPERTY_NAME = "options";
-
-        [property: JsonPropertyName(CONNSTRING_PROPERTY_NAME)]
-        public string ConnectionString { get; set; } = string.Empty;
-        public CosmosDbNoSqlOptions? CosmosDbNoSql { get; set; }
-        public CosmosDbPostgreSqlOptions? CosmosDbPostgreSql { get; set; }
-        public MsSqlOptions? MsSql { get; set; }
-        public PostgreSqlOptions? PostgreSql { get; set; }
-        public MySqlOptions? MySql { get; set; }
-
-        /// <summary>
-        /// Method to populate the database specific options from the "options"
-        /// section in data-source.
-        /// </summary>
-        public void PopulateDbSpecificOptions()
+        if (typeof(TOptionType).IsAssignableFrom(typeof(CosmosDbNoSQLDataSourceOptions)))
         {
-            if (DbOptions is null)
-            {
-                return;
-            }
-
-            switch (DatabaseType)
-            {
-                case DatabaseType.cosmosdb_nosql:
-                    CosmosDbNoSql = ((JsonElement)DbOptions).Deserialize<CosmosDbNoSqlOptions>(RuntimeConfig.SerializerOptions)!;
-                    break;
-                case DatabaseType.mssql:
-                    MsSql = ((JsonElement)DbOptions).Deserialize<MsSqlOptions>(RuntimeConfig.SerializerOptions)!;
-                    break;
-                case DatabaseType.mysql:
-                    MySql = ((JsonElement)DbOptions).Deserialize<MySqlOptions>(RuntimeConfig.SerializerOptions)!;
-                    break;
-                case DatabaseType.postgresql:
-                    PostgreSql = ((JsonElement)DbOptions).Deserialize<PostgreSqlOptions>(RuntimeConfig.SerializerOptions)!;
-                    break;
-                case DatabaseType.cosmosdb_postgresql:
-                    CosmosDbPostgreSql = ((JsonElement)DbOptions).Deserialize<CosmosDbPostgreSqlOptions>(RuntimeConfig.SerializerOptions)!;
-                    break;
-                default:
-                    throw new Exception($"DatabaseType: ${DatabaseType} not supported.");
-            }
+            return (TOptionType)(object)new CosmosDbNoSQLDataSourceOptions(
+                    Database: ReadStringOption("database"),
```

Comment:
For properties like `database`, `container`, `schema` etc., we can use constants in the `CosmosDbNoSQLDataSourceOptions` class.